lib/presentation/auth/auth_screen.dart

```diff
@@ -0,0 +1,156 @@
+/*
+ * TODO:
+ * - Abstract into widgets --MOST IMPORTANT FOR MERGE
+ * - New account flow --MOST IMPORTANT FOR MERGE
+ * - Found account flow --CAN BE DONE IN SEPARATE TASK
+ * - Not whitelisted number flow --CAN BE DONE IN SEPARATE TASK
+ * - Invite friends flow --CAN BE DONE IN SEPARATE TASK
+ */
+import 'package:auto_route/auto_route.dart';
+import 'package:collaction_app/infrastructure/core/injection.dart';
+import 'package:dots_indicator/dots_indicator.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+import '../../application/auth/auth_bloc.dart';
+import '../routes/app_routes.gr.dart';
+import '../shared_widgets/custom_app_bars/custom_appbar.dart';
+import '../themes/constants.dart';
+import '../utils/context.ext.dart';
+import 'pages/enter_username.dart';
+import 'pages/profile_photo.dart';
+import 'pages/verification_code.dart';
+import 'pages/verify_phone.dart';
+
+class AuthPage extends StatefulWidget {
+  const AuthPage({Key? key}) : super(key: key);
+
+  @override
+  _AuthPageState createState() => _AuthPageState();
+}
+
+class _AuthPageState extends State<AuthPage> {
+  // All Pages
+  final _pageController = PageController();
+  double _currentPage = 0.0;
+  late List<Widget> _pages;
+
+  bool _displayDots = true;
+
+  // Page One
+  // final GlobalKey<VerifyPhonePageState> _verifyPhoneKey = GlobalKey();
+
+  // Page Two
+
+  @override
+  void initState() {
+    super.initState();
+    // All pages
+    _pages = [
+      const VerifyPhonePage(/*key: _verifyPhoneKey*/),
+      const EnterVerificationCode(),
+      const EnterUserName(),
+      SelectProfilePhoto(
+        onSkip: () => _authDone(context),
+      )
+    ];
+
+    _pageController.addListener(() {
+      setState(() {
+        _currentPage = _pageController.page!;
+      });
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocProvider(
+      create: (context) => getIt<AuthBloc>(),
+      child: BlocListener<AuthBloc, AuthState>(
+        listener: (context, state) {
+          state.maybeMap(
+              smsCodeSent: (_) => _toPage(1),
+              loggedIn: (loggedInState) {
+                if (loggedInState.isNewUser) {
+                  // TODO - Handle if is existing user
+                } else {
+                  _toPage(2);
+                }
+              },
+              verificationCompleted: (_) {},
```

Comment:
Why is this empty?

Reply:
Removed it, it was being handled in the Pin input page